compiler/resolution/functionResolution.cpp

```diff
@@ -2584,18 +2584,11 @@ filterConcreteCandidate(Vec<ResolutionCandidate*>& candidates,
   }
 
   //
-  // If the candidate is an instantiation of a generic function,
-  // re-evaluate the where clause here in order to make sure it
-  // evaluates to true and to catch errors in it.  Ultimately, we'd
-  // like to do this for all concrete functions (instantiated or not),
-  // but at present, enabling it for them causes problems, perhaps
-  // because we're generating bogus where clauses on concrete
-  // functions today and getting away with it since they're ignored.
+  // Evaluate non-generic where clauses (also re-evaluates where cluases for
+  // instantiated generic functions.)
```

Comment:
At one point, it sounded like you had a branch that could nearly remove the where clause evaluation from the generic branch, which continues to seem attractive to me.  Are you leaving that for a separate branch or dropping the effort on the floor?